### ISSUE TITLE
Using regexes in alias names

### ISSUE THREAD

**Issue by tosti007** [roosta/i3wsr#16]
Hey there,

I would like to request the option to use regexes for alias names. This could make matching window names easier/more future proof for updates of said window names.

An example: 
Currently [gimp](https://www.gimp.org/) has the window name `gimp-2.10`, now if gimp has an update to v2.11 I will need to update my config file. I would rather like to define my alias as a regex, so that I can use `gimp-\d\.\d\d` or `^gimp`.

I foresee a few breaking changes/issues with this, being:
- Previous aliases such as `"Org.gnome.Nautilus"` need to be changed to `"Org\.gnome\.Nautilus"`, as the `.` otherwise means any character, but it will still work.
- Plainly using `substring` will also match against `somethingsubstring-something`, whilst it did not before. This should be changed to `^substring$` to obtain the current behavior.

**Comment by roosta** [comment]
Hi again, wanted to let you know I've made an initial implementation for alias regex matching.
the syntax is a little different from what you proposed due to string escapes, for for example the gimp pattern would look like this in the config:
```toml
"Gimp-\\d\\.\\d\\d" = "Gimp"
```
I decided against having regex as a toggle, since the substring matching works 99% of the time. I'll have a notice in the the new draft if this gets merged.

Bear in mind that this update builds on the `1.4.0` branch which has changed `no_names` option behavior, and introduce a new option called `no_icon_names` which behaves like the previous `no_names` (sorry about the confusion here but I thoughtlessly based the regex impl on that branch)

try out the [regex](https://github.com/roosta/i3wsr/tree/regex) branch if you got the time. Thanks.
EDIT: spelling
